### PR TITLE
Devtools update

### PR DESCRIPTION
**Pull request by bmhan12**
This PR:
- Rebuilds devtools with jquery dependency to fix local documentation builds (needed as part of RTD fix #1200)
  - `make docs` fails without this dependency:

```
Built target axom_doxygen
Built target doxygen_docs
Building HTML documentation with Sphinx for axom_docs target
 
Extension error:
Could not import extension sphinxcontrib.jquery (exception: No module named 'sphinxcontrib.jquery')
 
make[3]: *** [CMakeFiles/axom_docs] Error 2
make[2]: *** [CMakeFiles/axom_docs.dir/all] Error 2
make[1]: *** [CMakeFiles/docs.dir/rule] Error 2
make: *** [docs] Error 2
```

- Generates new host-configs for rzgenie post-toss4 update.

### CODE REVIEW

File: scripts/spack/packages/axomdevtools/package.py [hunk 1]
```diff
@@ -18,4 +18,5 @@ class Axomdevtools(BundlePackage):
     depends_on("graphviz")
     depends_on("py-sphinx")
     depends_on("py-shroud")
+    depends_on("py-sphinxcontrib-jquery")
```
Comment (bmhan12) [inline scripts/spack/packages/axomdevtools/package.py:21]: This is the main change of the PR, adding jquery as part of devtools recipe.

**Review by rhornung67 — APPROVED** [review]
Thank you @bmhan12

File: host-configs/lassen-blueos_3_ppc64le_ib_p9-clang@10.0.1.1.cmake [hunk 2]
```diff
@@ -79,31 +79,31 @@ set(BLT_CMAKE_IMPLICIT_LINK_DIRECTORIES_EXCLUDE "/usr/tce/packages/gcc/gcc-4.9.3
...
 set(LUA_DIR "${TPL_ROOT}/lua-5.4.4-o6kjgr3nt6ssvrj3mmonawxj3sbbcnel" CACHE PATH "")
 
-set(RAJA_DIR "${TPL_ROOT}/raja-2023.06.0-r2xeiifolea4xkrfluvkeehaynhvaipr" CACHE PATH "")
+set(RAJA_DIR "${TPL_ROOT}/raja-2023.06.0-jra4z7ctml3mpzsuixtbegawg7g75cec" CACHE PATH "")
 
-set(UMPIRE_DIR "${TPL_ROOT}/umpire-2023.06.0-x4j2bn55oz5hlsbgawepoej3kc3s3ama" CACHE PATH "")
+set(UMPIRE_DIR "${TPL_ROOT}/umpire-2023.06.0-h6wvcftrmg6w5e7etumawlsf5trgvsl7" CACHE PATH "")
 
-set(CAMP_DIR "${TPL_ROOT}/camp-2023.06.0-55emwjpfri4sr52rqvnfb5fiqitgkyml" CACHE PATH "")
+set(CAMP_DIR "${TPL_ROOT}/camp-2023.06.0-abpqpyhwsuhzu4csmdy2nejqzrkfqh7a" CACHE PATH "")
```
Comment (kennyweiss) [inline host-configs/lassen-blueos_3_ppc64le_ib_p9-clang@10.0.1.1.cmake:98]: Minor: Does anyone have a sense about why some of the TPL hashes changed, but not others? 

(In this host-config an others; it doesn't appear to be consistent, e.g. the conduit hash changed here but not in the next file -- `lassen-blueos-...-clang@10.0.1.2_cuda`)

File: scripts/spack/configs/toss_4_x86_64_ib/spack.yaml [hunk 3]
```diff
@@ -298,11 +298,11 @@ spack:
       - spec: cppcheck@2.9
         prefix: /collab/usr/gapps/axom/devtools/toss_4_x86_64_ib/latest/cppcheck-2.9
     doxygen:
-      version: [1.8.14]
+      version: [1.9.6]
```
Comment (kennyweiss) [inline scripts/spack/configs/toss_4_x86_64_ib/spack.yaml:301]: @bmhan12 -- was the doxygen upgrade necessary or just maintenance?

**Review by kennyweiss — APPROVED** [review]
Thanks @bmhan12 

Note: This also moves our rzgenie host-configs to toss4!

File: scripts/spack/configs/toss_4_x86_64_ib/spack.yaml (hunk 3 above)
Comment (bmhan12) [reply scripts/spack/configs/toss_4_x86_64_ib/spack.yaml:301]: > @bmhan12 -- was the doxygen upgrade necessary or just maintenance?
 
Maintenance-ish. Spack's highest default version changed for doxygen, so this bump was necessary for host-config setup.

**Review by agcapps — APPROVED** [review]

**Pull request merged**